js/src/redux/providers/certifications/middleware.js

```diff
@@ -14,38 +14,78 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import Contracts from '~/contracts';
-import { addCertification } from './actions';
+import { uniq } from 'lodash';
 
-const knownCertifiers = [ 'smsverification' ];
+import ABI from '~/contracts/abi/certifier.json';
+import Contract from '~/api/contract';
+import Contracts from '~/contracts';
+import { addCertification, removeCertification } from './actions';
 
 export default class CertificationsMiddleware {
   toMiddleware () {
+    const api = Contracts.get()._api;
+    const badgeReg = Contracts.get().badgeReg;
+    const contract = new Contract(api, ABI);
+    const Confirmed = contract.events.find((e) => e.name === 'Confirmed');
+    const Revoked = contract.events.find((e) => e.name === 'Revoked');
+
+    let certifiers = [];
+    let accounts = []; // these are addresses
+
+    const fetchConfirmedEvents = (dispatch) => {
+      if (certifiers.length === 0 || accounts.length === 0) return;
+      api.eth.getLogs({
+        fromBlock: 0,
+        toBlock: 'latest',
+        address: certifiers.map((c) => c.address),
+        topics: [ [ Confirmed.signature, Revoked.signature ], accounts ]
+      })
+        .then((logs) => contract.parseEventLogs(logs))
+        .then((logs) => {
+          logs.forEach((log) => {
+            const certifier = certifiers.find((c) => c.address === log.address);
+            if (!certifier) throw new Error(`Could not find certifier at ${log.address}.`);
+            const { id, name, title, icon } = certifier;
+            if (log.event === 'Revoked') {
```

Comment:
Really picky, but would prefer a line before this block, just indicates a break between assigning and doing. (Not a crisis, just aids readability once again)